rca/projects/utils.py

```diff
@@ -0,0 +1,28 @@
+def format_projects_for_gallery(projects):
+    """Internal method for formatting related projects to the correct
+    structure for the gallery template
+
+    Arguments:
+        projects: Queryset of project pages to format
+
+    Returns:
+        List: Maximum of 8 projects
+    """
+    items = []
+    for page in projects[:8]:
+        page = page.specific
+        meta = ""
+        related_school = page.related_school_pages.first()
```

Comment:
As this is being run for each project it might be worth using `prefetch_related` before the `for`. I _think_ this will work: `projects = projects.prefetch_related('related_school_pages__page')`

Reply:
Thanks, I've added that here https://github.com/torchbox/rca-wagtail-2019/pull/510/files#diff-e80c72762e3838277abb477451d1bbed7a074724f208ce368d6bc9857569704dR12